File: app/src/main/java/com/jeksonshar/jweather/repository/RepositoryProvider.java
package com.jeksonshar.jweather.repository;

import android.content.Context;
import android.content.SharedPreferences;

import com.jeksonshar.jweather.repository.room.RoomRepository;

public class RepositoryProvider {

    private static Repository instanceListWeather;

    private static SharedPreferencesRepository instanceLastUpdate;

    private RepositoryProvider() {
    }

    public static Repository getInstanceListWeather(Context context) {
        if (instanceListWeather == null) {
            instanceListWeather = new RoomRepository(context);
        }
        return instanceListWeather;
    }

    public static SharedPreferencesRepository getInstanceLastUpdate(Context context) {
        if (instanceLastUpdate == null) {
            instanceLastUpdate = new SharedPreferencesRepository(context);
        }
        return instanceLastUpdate;
    }
}

File: app/src/main/java/com/jeksonshar/jweather/repository/RepositoryProvider.kt
package com.jeksonshar.jweather.repository

import android.content.Context
import com.jeksonshar.jweather.repository.room.RoomRepository
//TODO лушче сделать application статикой и оттуда тянуть твои зависимости
object RepositoryProvider {

    private var instanceListWeather: RepositoryDataBase? = null
    private var instanceLastUpdate: SharedPreferencesRepository? = null

    fun getInstanceListWeather(context: Context?): RepositoryDataBase? {
        if (instanceListWeather == null) {
            instanceListWeather = RoomRepository(context)
        }
        return instanceListWeather
    }

    fun getInstanceLastUpdate(context: Context?): SharedPreferencesRepository? {
        if (instanceLastUpdate == null) {
            instanceLastUpdate = SharedPreferencesRepository(context!!)
        }
        return instanceLastUpdate
    }
}